iceoryx_posh/test/moduletests/test_roudi.cpp

```diff
@@ -0,0 +1,545 @@
+        m_roudiComponents->m_rouDiMemoryManager,
+        m_roudiComponents->m_portManager,
+        RouDi::RoudiStartupParameters{roudi::MonitoringMode::ON, false, RouDi::MQThreadStart::DEFER_START}));
+
+    m_roudiApp->startMQThreadTest();
+
+    EXPECT_DEATH(m_roudiApp->startMQThreadTest(), ".*");
+
+    m_roudiApp->shutDownTest();
+}
+
+TEST_F(RouDiBaseClass_test, ShutDownKillProcessInDestructorTrueIsSuccessfull)
+{
+    std::unique_ptr<IceOryxRouDiComponents> m_roudiComponents =
+        std::unique_ptr<IceOryxRouDiComponents>(new IceOryxRouDiComponents(iox::RouDiConfig_t().setDefaults()));
+    std::unique_ptr<RoudiClassTest> m_roudiApp = std::unique_ptr<RoudiClassTest>(
+        new RoudiClassTest(m_roudiComponents->m_rouDiMemoryManager,
+                           m_roudiComponents->m_portManager,
+                           RouDi::RoudiStartupParameters{roudi::MonitoringMode::ON, true}));
+    MqMessage message;
+    message << mqMessageTypeToString(MqMessageType::CREATE_PUBLISHER) << "AppName"
+            << "123123"
+            << "456"
+            << "789"
+            << "123";
+    auto cmd = runtime::stringToMqMessageType(message.getElementAtIndex(0).c_str());
+    std::string processName = message.getElementAtIndex(1);
+
+    m_roudiApp->processMessageTest(message, cmd, ProcessName_t(cxx::TruncateToCapacity, processName));
+
+    m_roudiApp->shutDownTest();
+}
+
+TEST_F(RouDiBaseClass_test, ShutDownCalledTwoTimesIsSuccessfull)
+{
+    std::unique_ptr<IceOryxRouDiComponents> m_roudiComponents =
+        std::unique_ptr<IceOryxRouDiComponents>(new IceOryxRouDiComponents(iox::RouDiConfig_t().setDefaults()));
+    std::unique_ptr<RoudiClassTest> m_roudiApp = std::unique_ptr<RoudiClassTest>(
+        new RoudiClassTest(m_roudiComponents->m_rouDiMemoryManager,
+                           m_roudiComponents->m_portManager,
+                           RouDi::RoudiStartupParameters{roudi::MonitoringMode::ON, true}));
+
+    m_roudiApp->shutDownTest();
+
+    m_roudiApp->shutDownTest();
+}
+
+TEST_F(RouDiBaseClass_test, SendCorrectRegMessagetoRoudiReturnSuccessfullySent)
+{
+    const iox::ProcessName_t m_runtimeName{"App"};
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    PoshRuntime* m_runtime{&iox::runtime::PoshRuntime::initRuntime(m_runtimeName)};
+
+    MqMessage message;
+    MqMessage m_receiveBuffer;
+    const iox::NodeName_t m_nodeName{"testNode"};
+    message << mqMessageTypeToString(MqMessageType::REG) << "App" << std::to_string(getpid())
+            << std::to_string(posix::PosixUser::getUserOfCurrentProcess().getID()) << std::to_string(0)
+            << static_cast<cxx::Serialization>(version::VersionInfo::getCurrentVersion()).toString();
+
+    const auto successfullySent = m_runtime->sendRequestToRouDi(message, m_receiveBuffer);
+
+    EXPECT_TRUE(m_receiveBuffer.isValid());
+    EXPECT_TRUE(successfullySent);
+}
```

Comment:
this is a `PoshRuntime` test

Reply:
That was my only way to be able to get an assertion in this test.

Reply:
But that still makes it a runtime test. You basically combined the `SendRequestToRouDiValidMessage` and `ValidAppName` from `PoshRuntime_test`.
Like I already said, this class is currently not easy testable and needs refactoring before useful tests can be written. On the positive side, much of this is already tested by other means, e.g. the posh runtime tests which uses the RouDiEnvironment which uses this class.

Reply:
Sure, but what I did could serve as a base for when the class will be refactored. By using poshruntime, I am able to improve the coverage and add Assert. Sure, it will be change after refactoring but currently, it is the only thing I can do.

Reply:
But this doesn't increase the coverage in a meaningful way. In this specific case there is nothing tested which isn't already tested in `PoshRuntime_test`. Other tests which don't have any assert give a false sense of safety because they give the illusion of increased coverage while they cannot fail, except when they crash the application but this is already ensured by other means.
This also makes refactoring harder because not only the code needs to be refactored but also the tests, which would be okay if the tests did something meaningful. The chances are good that they will be completely rewritten.

Please don't write tests where nothing can be asserted just to hunt a coverage number. In general, if nothing can be asserted either the class needs to be adjusted or no test should be written.

Reply:
I know that the tests without assert are kind of useless right now. But, I thought of using them after refactoring as the logic, from my side, should be the same.

Reply:
But that is not necessarily the case and it makes things harder to refactor without having any benefit right now.

Reply:
I see, i will just checkout the change so.